### ISSUE TITLE
`ignore_errors` doesn't work as expected if `show_indicator` is set to False

### ISSUE THREAD

**Issue by adayush** [confident-ai/deepeval#809]
**Description**
From what I understood, if `ignore_errors` is set to True then if any TestCase's metric measurement fails it would not affect the evaluation of other TestCases and their evaluation would continue.

https://github.com/confident-ai/deepeval/blob/7032d7d48bc6fa53d1562979e97d0dda94cb571c/deepeval/metrics/indicator.py#L131-L141

But this only happens when `show_indicator` is set to True. When it is set to false, the `ignore_errors` flag is of no use as the errors do affect and stop the evaluation of all TestCases.

https://github.com/confident-ai/deepeval/blob/7032d7d48bc6fa53d1562979e97d0dda94cb571c/deepeval/metrics/indicator.py#L171-L179

The error occurs in `metric.a_measure` and not in `tasks.append` and thus it is not caught by the try except blocks.

**To Reproduce**

1. Use a custom metric and raise an Exception in a_measure
2. Call evaluate with `show_indicator` to False and `ignore_errors` set to True

```
    results = evaluate(
            test_cases=test_cases,
            metrics=list(evaluators),
            show_indicator=False,
            print_results=False,
            use_cache=False,
            ignore_errors=True
        )
```

**Desktop:**
 - OS: Docker with Ubuntu
 - Version 0.21.48

**Comment by penguine-ip** [comment]
@adayush thanks for the detailed explanation, was easy to reproduce! Fixed it in v0.21.50, TYSM!

**Comment by adayush** [comment]
Glad to know I was able to help! 😄
